docs/cunumeric/source/comparison/_comparison_generator.py

```diff
@@ -1,44 +1,62 @@
 import importlib
 
+import numpy
+
+blocklist = [
+    "test",
+    "add_docstring",
+    "abs",
+    "add_newdoc",
+    "add_newdoc_ufunc",
+    "alen",
+    "alltrue",
+    "bitwise_not",
+    "compare_chararrays",
+    "cumproduct",
+    "fastCopyAndTranspose",
+    "get_array_wrap",
+    "iterable",
+    "max",
+    "min",
+    "ndim",
+    "product",
+    "recfromcsv",
+    "recfromtxt",
+    "round",
+    "safe_eval",
+    "set_numeric_ops",
+    "size",
+    "sometrue",
+    "loads",
+    "mafromtxt",
+    "matmul",
+    "ndfromtxt",
+]
+
 
 def _filter(obj, n):
     try:
         return (
-            n
-            not in [
-                "test",
-                "add_docstring",
-                "abs",
-                "add_newdoc",
-                "add_newdoc_ufunc",
-                "alen",
-                "alltrue",
-                "bitwise_not",
-                "compare_chararrays",
-                "cumproduct",
-                "fastCopyAndTranspose",
-                "get_array_wrap",
-                "iterable",
-                "max",
-                "min",
-                "ndim",
-                "product",
-                "recfromcsv",
-                "recfromtxt",
-                "round",
-                "safe_eval",
-                "set_numeric_ops",
-                "size",
-                "sometrue",
-                "loads",
-                "mafromtxt",
-                "matmul",
-                "ndfromtxt",
-            ]  # not in blocklist
+            n not in blocklist
+            and callable(getattr(obj, n))  # callable
+            and not isinstance(getattr(obj, n), type)  # not class
+            and n[0].islower()  # starts with lower char
+            and not n.startswith("__")  # not special methods
+            and not isinstance(getattr(obj, n), numpy.ufunc)
+        )
+    except:  # noqa: E722
+        return False
+
+
+def _filter_ufunc(obj, n):
```

Comment:
I guess you could factor out the common part of this and `_filter` even further. the two are different only in the last predicate.

Reply:
Sure, I will try